components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/util/EventPayloadCryptographyUtils.java

```diff
@@ -0,0 +1,221 @@
+
+import com.nimbusds.jose.EncryptionMethod;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWEAlgorithm;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.source.DefaultJWKSetCache;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jose.util.DefaultResourceRetriever;
+import com.nimbusds.jose.util.Resource;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.identity.event.websubhub.publisher.internal.WebSubHubAdapterDataHolder;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ASYMMETRIC_ENCRYPTION_ALGORITHM;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.CRYPTO_KEY_JSON_KEY;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.CRYPTO_KEY_RESPONSE_JSON_KEY;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ENCRYPTED_PAYLOAD_JSON_KEY;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ENCRYPTION_KEY_ENDPOINT_URL_TENANT_PLACEHOLDER;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ErrorMessages.ERROR_RETRIEVING_ENCRYPTION_PUBLIC_KEY;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.IV_PARAMETER_SPEC_JSON_KEY;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.SYMMETRIC_ENCRYPTION_ALGORITHM;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.SYMMETRIC_ENCRYPTION_ALGORITHM_WITH_MODE;
+
+/**
+ * This class contains utility methods for encrypting Event Payloads.
+ */
+public class EventPayloadCryptographyUtils {
```

Comment:
Do we use this in our current implementation ?

Reply:
https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/c70da89e5b94a72ca32cf8008995faf468357425